lib/proxy/helpers.rb

```diff
@@ -25,7 +25,8 @@ def log_halt code=nil, exception=nil
 
   # read the HTTPS client certificate from the environment and extract its CN
   def https_cert_cn
-    certificate_raw = request.env['SSL_CLIENT_CERT'].to_s
+    ssl_client_cert_header = Proxy::SETTINGS.ssl_client_cert_header || 'SSL_CLIENT_CERT'
+    certificate_raw = request.env[ssl_client_cert_header].to_s.gsub(/   /, "\n")
```

Comment:
In my environment, nginx/1.8.1, Phusion Passenger 5.0.26, I was getting  `nested asn1 error` while parsing SSL_CLIENT_CERT. it was because somehow, new lines was replaced with spaces. I reverted that, to valid PEM format. It will not affect standalone running service. I know it's not a best solution, it would be better to implement what I suggested in [issue](http://projects.theforeman.org/issues/14153).